lib/autoload/send_file.rb

```diff
@@ -16,11 +31,16 @@ def self.send_file(file, public_name = File.basename(file))
     Addressable::URI.encode(File.join('/', downloads_dir, temporary_dir, public_name))
   end
 
-  # Obtains the local path for the given publicly accessible file path.
-  #
-  # @param [String] path The URL to the publicly accessible file.
-  # @return [String] The absolute file path to the publicly accessible file.
-  def self.local_path(path)
-    File.join(Rails.public_path, URI.decode_www_form_component(path))
+  def self.send_file_production(file, public_name)
+    current_time = Time.now.to_i
+    s3_key = "downloads/#{current_time}/#{public_name}"
+    object = S3_BUCKET.object(s3_key)
+    File.open(file, 'rb') do |f|
+      object.put(body: f)
+    end
+
+    signer = Aws::S3::Presigner.new
+    url, = signer.presigned_request(:get_object, bucket: S3_BUCKET.name, key: s3_key)
```

Comment:
There's no need to use the headers?

Reply:
We do not set any headers when generating the url as there is no specific use case I can think of as of now. As such, the headers returned is empty. I just found that there's another method that can be used `presigned_url` that only returns the url.